### PR TITLE
Alert service mvp

### PR DESCRIPTION
**Pull request by jarrydlee**
fixes https://github.com/PagerNation/PagerNation/issues/81

### CODE REVIEW

File: server/services/group.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ function createGroup(groupObject) {
 
   return new Promise((resolve, reject) => {
     Joi.validate(groupObject, groupSchema, (err, value) => {
+      console.log(err, value);
```
Comment (jarrydlee) [inline server/services/group.js]: Remove

File: server/services/ticket.js [hunk 2]
```diff
@@ -29,22 +30,24 @@ function createTicket(ticketDetails) {
...
-        reject(err);
-      } else {
-        resolve(Ticket.create(value));
+        return reject(err);
       }
+      resolve(Ticket.create(value));
     });
+  })
+  .then((ticket) => {
+    // alertService.createAlert(ticket);
```
Comment (jarrydlee) [inline server/services/ticket.js]: Commented out right now until I add in `.catch()` logic to stop tests from failing.

File: server/tests/models/group.test.js [hunk 3]
```diff
@@ -52,23 +52,4 @@ describe('## Groups', () => {
       });
     });
   });
-
-  context('# when users is not there', () => {
```
Comment (jarrydlee) [inline server/tests/models/group.test.js]: Users array has a default now, test no longer needed

File: server/tests/services/phone.test.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    before((done) => {
+      build('device', fixtures.smsDevice({ contactInformation: '+15005550006' }))
+        .then((device) => {
+          smsDevice = device;
+          done();
+        });
+    });
+
+    it('should successfully send a sms message', (done) => {
+      phoneService.sendMessage(null, null, smsDevice)
```
Comment (paradime) [inline server/tests/services/phone.test.js]: Does this actually send the message?

Comment (jarrydlee) [inline server/tests/services/phone.test.js]: Nope, it uses their [test credentials](https://www.twilio.com/docs/api/rest/test-credentials) to return specific responses based on the 'to' phone number

File: config/env/development.js [hunk 5]
```diff
@@ -1,6 +1,14 @@
...
-  port: 3000
+  port: 3000,
+  email: {
+    user: process.env.EMAIL_USER,
+    pass: process.env.EMAIL_PASS
+  },
+  twilio: {
+    accountSid: process.env.PHONE_SID,
+    token: process.env.PHONE_TOKEN,
+    fromPhone: '+15854818574'
```
Comment (kalebdavis) [inline config/env/development.js]: Whose number is this?

File: server/models/group.js [hunk 6]
```diff
@@ -23,7 +23,8 @@ const GroupSchema = new mongoose.Schema({
     type: [{
       type: mongoose.Schema.Types.ObjectId,
```
Comment (kalebdavis) [inline server/models/group.js]: This should be a string, I definitely thought I changed that earlier.

File: server/models/ticket.js [hunk 7]
```diff
@@ -3,13 +3,20 @@ import httpStatus from 'http-status';
...
-    type: mongoose.Schema.Types.ObjectId,
+  groupName: {
+    type: String,
     required: true,
   },
   metadata: mongoose.Schema.Types.Mixed
 });
 
+TicketSchema.virtual('group', {
+  ref: 'Group', // The model to use
```
Comment (kalebdavis) [inline server/models/ticket.js]: Nit: can you make these comments line up with each other?

File: server/tests/services/phone.test.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+describe('## Phone Service', () => {
+  describe('# Call', () => {
+    const INVALID_NUMBER = '+15005550001';
+    const INVALID_NUMBER_ERROR_CODE = 21217;
+
+    let phoneDevice;
+
+    before((done) => {
+      build('device', fixtures.phoneDevice({ contactInformation: '+15005550006' }))
```
Comment (kalebdavis) [inline server/tests/services/phone.test.js]: Pull this number into a constant bc it's used more than once

**Comment by kalebdavis** [comment]
LGTM

**Pull request merged**